### ISSUE TITLE
02. Add regression tests for gated memory-target fallback

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#441]
# 02. Add regression tests for gated memory-target fallback

## Background

After fixing target selection, add tests that fail on the current implementation and pass only when the criterion chooses the best eligible candidate after gates are applied.

The relevant test file is:

```text
packages/seisai-engine/tests/test_fbpick_coarse_trace_memory_consistency.py
```

## Required tests

Add focused tests for at least these cases.

### 1. Confidence-gate fallback

Setup:

```text
current row: primary=0, secondary=0, raw_trace_index=10
target_policy: primary_cross
target_min_pmax: 0.01
```

Memory entries:

```text
slot A: primary=1, secondary=0, count high, pmax=0.001, pick_raw=1.0
slot B: primary=2, secondary=0, count lower, pmax=0.900, pick_raw=3.0
```

Expected:

```text
- the row is used
- low_pmax_reject_rows records the rejected candidate or otherwise the test verifies fallback behavior
- loss is positive/finite
```

The important assertion is that a valid lower-ranked candidate is used instead of dropping the row because slot A failed `target_min_pmax`.

### 2. Age-gate fallback

Setup:

```text
target_policy: primary_cross
max_age_batches: small value, e.g. 2
current batch step/index: sufficiently larger than one stored slot's last_batch
```

Memory entries:

```text
slot A: primary=1, count high, last_batch old enough to fail max_age_batches
slot B: primary=2, count lower, last_batch recent enough to pass
```

Expected:

```text
- the row is used
- the selected target comes from the recent valid slot
```

### 3. Label-guard fallback

Setup:

```text
target_policy: primary_cross or primary_any
label_guard_samples: 1
fb_idx_view: 2
```

Memory entries:

```text
slot A: higher-ranked candidate, pick_raw projects to a view index far from 2
slot B: lower-ranked candidate, pick_raw projects to view index 2 or within guard
```

Expected:

```text
- the row is used
- label guard rejects slot A but does not cause the whole row to be dropped
- the final target is slot B
```

### 4. No-eligible-candidate remains backward-safe

Keep or extend the existing no-target test so that when all candidates fail gates, the consistency loss remains finite and backward-safe:

```text
loss.backward() works
logits.grad exists
no NaN/Inf
```

## Suggested helper additions

It is acceptable to add small test helpers to avoid direct repetition:

```python
def _put_memory_entry(bank, *, raw=10, primary, secondary, pick_raw, pmax, count, last_batch): ...
```

Avoid overfitting tests to private implementation details unless necessary. Prefer assertions on:

```text
trace_memory/used_rows
trace_memory/low_pmax_reject_rows
trace_memory/age_reject_rows
trace_memory/label_guard_reject_rows
loss finite / loss > base loss
```

If the production implementation exposes selected source ids for diagnostics, asserting the selected source id is also acceptable.

## Acceptance criteria

- Tests fail against the current one-slot-before-gates implementation.
- Tests pass after issue 01 is implemented.
- Tests cover `primary_cross` at minimum.
- Existing PR3 tests remain valid.
